### PR TITLE
add check-before-deployment.md

### PR DESCRIPTION
**Pull request by superlzs0476**
### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### CODE REVIEW

File: check-before-deployment.md [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+
+    ```bash
+    sudo yum install ntp ntpdate && \
+    sudo systemctl start ntpd.service && \
+    sudo systemctl enable ntpd.service
+    ```
+
+## 手动配置 SSH 互信及 sudo 免密码
+
+本段介绍手动配置中控机到目标节点的互信配置实施方案，如果是通过 TiUP 工具部署的 TiDB 不需要考虑手工配置 SSH 互信及免密登陆。
```
Comment (kissmydb) [inline check-before-deployment.md:244]:
```suggestion
对于有需求，通过手动配置中控机至目标节点互信的场景，可参考本段。通常推荐使用 TiUP 部署工具会自动配置 SSH 互信及免密登陆。
```

File: check-before-deployment.md [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+    [root@10.0.1.1 tidb]#
+    ```
+
+## 安装 numactl 工具
+
+本段主要介绍如果安装 NUMA 工具。在线上环境中，因为硬件采购机器配置较高，为了更合理规划资源，会考虑单机多实例部署 TiDB 或者 TiKV。NUMA 绑核工具的使用，主要为了防止 CPU 资源的争抢，引发性能衰退。
+
+> **注意：**
+>
+> - numa 绑核是用来隔离 CPU 资源一种方法，适合高配置物理机环境部署多实例使用。
```
Comment (ran-huang) [inline check-before-deployment.md:305]:
```suggestion
> - NUMA 绑核是用来隔离 CPU 资源的一种方法，适合高配置物理机环境部署多实例使用。
```

File: check-before-deployment.md [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+    Usage:
+    cluster exec <cluster-name> [flags]
+
+    Flags:
+        --command string   the command run on cluster host (default "ls")
+    -h, --help             help for exec
+        --sudo             use root permissions (default false)
+    ```
+
+    操作命令，将 tidb-test 集群所有目标主机通过 sudo 权限执行安装命令
```
Comment (ran-huang) [inline check-before-deployment.md:336]:
```suggestion
    将 tidb-test 集群所有目标主机通过 sudo 权限执行安装命令
```

File: check-before-deployment.md [hunk 4]
```diff
@@ -0,0 +1,343 @@
...
+
+{{< copyable "shell-regular" >}}
+    
+```bash
+echo "vm.swappiness = 0">> /etc/sysctl.conf 
+swapoff -a && swapon -a
+sysctl -p
+```
+
+## 检测及关闭部署机器的防火墙
```
Comment (kissmydb) [inline check-before-deployment.md:126]:
```suggestion
## 检测及关闭目标部署机器的防火墙
```

File: check-before-deployment.md [hunk 5]
```diff
@@ -0,0 +1,343 @@
...
+    
+```bash
+echo "vm.swappiness = 0">> /etc/sysctl.conf 
+swapoff -a && swapon -a
+sysctl -p
+```
+
+## 检测及关闭目标部署机器的防火墙
+
+本段介绍如何关闭目标主机防火墙配置，因为在 TiDB 集群中，需要将节点间的访问端口打通才可以保证读写请求、数据心跳等信息的正常的传输。在普遍线上场景中，数据库到业务服务和数据库节点点网络联通都是在安全域内完成数据交互。如果没有特殊安全的要求，建议将目标节点的防火墙进行关闭。否则还是建议[按照端口使用规则](/hardware-hardware-and-software-requirements.md)，将端口信息配置到防火墙服务的白名单中。
```
Comment (kissmydb) [inline check-before-deployment.md:128]:
```suggestion
本段介绍如何关闭目标主机防火墙配置，因为在 TiDB 集群中，需要将节点间的访问端口打通才可以保证读写请求、数据心跳等信息的正常的传输。在普遍线上场景中，数据库到业务服务和数据库节点的网络联通都是在安全域内完成数据交互。如果没有特殊安全的要求，建议将目标节点的防火墙进行关闭。否则建议[按照端口使用规则](/hardware-hardware-and-software-requirements.md)，将端口信息配置到防火墙服务的白名单中。
```

File: check-before-deployment.md [hunk 6]
```diff
@@ -0,0 +1,342 @@
...
+
+## 在 TiKV 部署目标机器上添加数据盘 EXT4 文件系统挂载参数
+
+生产环境部署，建议使用 EXT4 类型文件系统的 NVME 类型的 SSD 磁盘存储 TiKV 数据文件 。这个配置方案为最佳实施方案，其可靠性、安全性、稳定性已经在大量线上场景中得到证实。
+
+> **注意：**
+>
+> 推荐 TiKV 部署目标机器的数据目录使用 EXT4 文件系统格式。相比于 XFS 文件系统格式，EXT4 文件系统格式在 TiDB 集群部署案例较多，生产环境优先选择使用 EXT4 文件系统格式。
+
+使用 `root` 用户登录目标机器，将部署目标机器数据盘格式化成 ext4 文件系统，挂载时添加 `nodelalloc` 和 `noatime` 挂载参数。`nodelalloc` 是必选参数，否则 TiUP 安装时检测无法通过；`noatime` 是可选建议参数。
```
[the remark this answers is not in the document]
Comment (kissmydb) [reply check-before-deployment.md:18]: 暂时先不用，越解释越复杂

File: check-before-deployment.md [hunk 7]
```diff
@@ -0,0 +1,342 @@
...
+    
+```bash
+echo "vm.swappiness = 0">> /etc/sysctl.conf 
+swapoff -a && swapon -a
+sysctl -p
+```
+
+## 检测及关闭部署机器的防火墙
+
+本段介绍如何关闭目标主机防火墙配置，因为在 TiDB 集群中，需要将节点间的访问端口打通才可以保证读写请求、数据心跳等信息的正常的传输。在普遍线上场景中，数据库到业务服务和数据库节点点网络联通都是在安全域内完成数据交互。如果没有特殊安全的要求，建议将目标节点的防火墙进行关闭。否则还是建议[按照端口使用规则](/hardware-hardware-and-software-requirements.md)，将端口信息配置到防火墙服务的白名单中。
```
[the remark this answers is not in the document]
Comment (kissmydb) [reply check-before-deployment.md]: 已修改，3ks

File: check-before-deployment.md [hunk 8]
```diff
@@ -0,0 +1,343 @@
...
+    
+```bash
+echo "vm.swappiness = 0">> /etc/sysctl.conf 
+swapoff -a && swapon -a
+sysctl -p
+```
+
+## 检测及关闭目标部署机器的防火墙
+
+本段介绍如何关闭目标主机防火墙配置，因为在 TiDB 集群中，需要将节点间的访问端口打通才可以保证读写请求、数据心跳等信息的正常的传输。在普遍线上场景中，数据库到业务服务和数据库节点的网络联通都是在安全域内完成数据交互。如果没有特殊安全的要求，建议将目标节点的防火墙进行关闭。否则建议[按照端口使用规则](/hardware-hardware-and-software-requirements.md)，将端口信息配置到防火墙服务的白名单中。
```
Comment (kissmydb) [inline check-before-deployment.md:128]:
```suggestion
本段介绍如何关闭目标主机防火墙配置，因为在 TiDB 集群中，需要将节点间的访问端口打通才可以保证读写请求、数据心跳等信息的正常的传输。在普遍线上场景中，数据库到业务服务和数据库节点的网络联通都是在安全域内完成数据交互。如果没有特殊安全的要求，建议将目标节点的防火墙进行关闭。否则建议[按照端口使用规则](/hardware-and-software-requirements.md)，将端口信息配置到防火墙服务的白名单中。
```

**Comment by ran-huang** [comment]
@zz-jason PTAL. Thanks!

File: check-before-deployment.md [hunk 9]
```diff
@@ -0,0 +1,343 @@
...
+
+    ```bash
+    fdisk -l
+    ```
+
+    ```
+    Disk /dev/nvme0n1: 1000 GB
+    ```
+
+2. 创建分区表。
```
Comment (zz-jason) [inline check-before-deployment.md:38]:
```suggestion
2. 创建分区。
```

**Comment by ran-huang** [comment]
@zz-jason PTAL

File: check-before-deployment.md [hunk 10]
```diff
@@ -0,0 +1,343 @@
+---
+title: TiDB 环境与系统配置检查
+category: how-to
+---
+
+# TiDB 环境与系统配置检查
+
+本章节将介绍部署 TiDB 前的环境检查操作，以下重要性参照文档顺序的优先级。
```
Comment (lilin90) [inline check-before-deployment.md:8]:
```suggestion
本文介绍部署 TiDB 前的环境检查操作，以下各项操作按优先级排序。
```

File: check-before-deployment.md [hunk 11]
```diff
@@ -0,0 +1,343 @@
+---
+title: TiDB 环境与系统配置检查
```
Comment (lilin90) [inline check-before-deployment.md:2]:
```suggestion
title: TiDB 环境与系统配置检查
summary: 了解部署 TiDB 前的环境检查操作。
```

File: check-before-deployment.md [hunk 12]
```diff
@@ -0,0 +1,344 @@
...
+    sudo -su root
+    ```
+
+    ```
+    [root@10.0.1.1 tidb]#
+    ```
+
+## 安装 numactl 工具
+
+本段主要介绍如果安装 NUMA 工具。在线上环境中，因为硬件采购机器配置较高，为了更合理规划资源，会考虑单机多实例部署 TiDB 或者 TiKV。NUMA 绑核工具的使用，主要为了防止 CPU 资源的争抢，引发性能衰退。
```
Comment (ran-huang) [inline check-before-deployment.md:303]: Confirmed with author: `因为硬件采购机器配置较高` -> sometimes our purchasing department has purchased much higher configurations than normal hardware hosts.

File: check-before-deployment.md [hunk 13]
```diff
@@ -0,0 +1,344 @@
...
+
+本文介绍部署 TiDB 前的环境检查操作，以下各项操作按优先级排序。
+
+## 在 TiKV 部署目标机器上添加数据盘 EXT4 文件系统挂载参数
+
+生产环境部署，建议使用 EXT4 类型文件系统的 NVME 类型的 SSD 磁盘存储 TiKV 数据文件。这个配置方案为最佳实施方案，其可靠性、安全性、稳定性已经在大量线上场景中得到证实。
+
+> **注意：**
+>
+> 推荐 TiKV 部署目标机器的数据目录使用 EXT4 文件系统格式。相比于 XFS 文件系统格式，EXT4 文件系统格式在 TiDB 集群部署案例较多，生产环境优先选择使用 EXT4 文件系统格式。
```
Comment (yikeke) [inline check-before-deployment.md:17]: 这一段可以删了，和上面一句话重复了 @kissmydb @ran-huang

Comment (ran-huang) [reply check-before-deployment.md:17]: Deleted in https://github.com/pingcap/docs-cn/pull/3457.

File: check-before-deployment.md (hunk 12 above)
Comment (ran-huang) [reply check-before-deployment.md:303]: fixed in https://github.com/pingcap/docs-cn/pull/3457

**Pull request merged**